franklin/github/views.py

```diff
@@ -172,6 +176,17 @@ def register_repo(request):
     # TODO - Lock this endpoint down so it's only callable from the future
     # admin panel.
     if request.method == 'POST':
+        """
+        # Calling github will look something like this
+        user = User.objects.get(...)
+        social = user.social_auth.get(provider='google-oauth2')
+        response = requests.get(
+                    'https://www.googleapis.com/plus/v1/people/me/people/visible',
+                        params={'access_token':
+                            social.extra_data['access_token']}
+                        )
+        friends = response.json()['items']
+        """
         serializer = SiteSerializer(data=request.data)
```

Comment:
Step 2 - list all of the users repos on the UI with a button called `activate` next to each.
Step 3 - clicking that button calls this endpoint for the user
Step 4 - We call github as the user and generate a `deploy_key` for the project being registered. 
Step 5 - $$